### PR TITLE
Remove getInferredAuth from RawConfiguration / embed auth source in AuthProperties

### LINKED ISSUE #1245: Refactor auth descriptors in plugins
Currently, plugin-specific parameter name strings are passed around using some methods in `RawConfiguration`, separately from the actual values used for authentication. It may be a good idea to introduce an `AuthProvider` interface to keep all these values together.

[Original comment.](https://github.com/GoogleContainerTools/jib/pull/1244#pullrequestreview-174087594)

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #1245.

Also taking care of the following TODO, removing `getInferredAuth` from `RawConfiguration`. With this, `RawConfiguration` doesn't need `EventHandler` when instantiating, which means now we can instantiated it before instantiating `MavenProjectProperites`.

```java
  // TODO: This is only for getting values from Maven settings.xml, and in some sense, auth info
  // from settings.xml is not necessary raw configuration values. Consider removing it.
  // https://github.com/GoogleContainerTools/jib/pull/1163#discussion_r228389684
  Optional<AuthProperty> getInferredAuth(String authTarget) throws InferredAuthRetrievalException;
```

### CODE REVIEW

File: jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/InferredAuthProvider.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.google.cloud.tools.jib.plugins.common;
```
Comment (coollog) [inline jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/InferredAuthProvider.java]: copyright

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java [hunk 2]
```diff
@@ -106,21 +105,22 @@ public void buildDocker()
...
-      RawConfiguration rawConfiguration = new GradleRawConfiguration(jibExtension);
 
       GradleHelpfulSuggestionsBuilder gradleHelpfulSuggestionsBuilder =
           new GradleHelpfulSuggestionsBuilder(HELPFUL_SUGGESTIONS_PREFIX, jibExtension);
 
       PluginConfigurationProcessor pluginConfigurationProcessor =
           PluginConfigurationProcessor.processCommonConfigurationForDockerDaemonImage(
-              rawConfiguration,
+              new GradleRawConfiguration(jibExtension),
+              ignored -> java.util.Optional.empty(),
```
[the remark this answers is not in the document]
Comment (chanseokoh) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java]: Unfortunately, this file is using the Gradle `Optional` annotation.

[the remark this answers is not in the document]
Comment (coollog) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java]: Ah okay

**Comment by TadCordle** [comment]
Btw this also fixes #1245, I think

**Pull request merged**